src/main/kotlin/org/jetbrains/research/transformations/Transformations.kt

```diff
@@ -0,0 +1,36 @@
+package org.jetbrains.research.transformations
+
+import com.github.gumtreediff.tree.ITree
+
+public class Transformations() {
+    public val var_anon : MutableMap<String, String> = mutableMapOf()
+    private var num_var_anon = 0
+
+    // this method should anonymize all variables in code
+    fun anonymize(tree :ITree, toStoreMetadata :Boolean) {
+        if (!tree.isLeaf) {
```

Comment:
I think you can not to check if it is a Leaf. If you look at gumtree implementation of this method
```
    public boolean isLeaf() {
        return getChildren().size() == 0;
    }
```
you'll see that in case it's a leaf it just has empty children list, so it won't be iterating anyway

Reply:
I am totally agree